scripts/psake-build.ps1

```diff
@@ -0,0 +1,190 @@
+    exec {
+        & $msbuildExe $srcDir\Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native\ /property:Configuration=$configuration /property:Platform=$targetPlatform
+    }
+}
+
+Task Build.SesClient -Depends Requires.MsBuild, Requires.Configuration, Requires.Platform {
+    exec {
+        & $msbuildExe $srcDir\sesclient.native\ /property:Configuration=$configuration /property:Platform=$targetPlatform
+    }
+}
+
+Task Unit.Tests -Depends Requires.DotNetExe, Build.SesTest {
+    foreach ($project in (resolve-path $testsDir\*\*.csproj)){
+        Write-SubtaskName (split-path $project -Leaf)
+        exec {
+            & $dotnetExe test $project
+        }
+    }
+}
+
+Task Publish.SesTest.Win64 -Depends Requires.DotNetExe, Restore.NuGetPackages {
+    exec {
+        & $dotnetExe publish $srcDir\sestest\sestest.csproj --self-contained --output $publishDir\sestest\win10-x64 --runtime win10-x64
+    }
+
+    exec {
+        compress-archive $publishDir\sestest\win10-x64\* $publishDir\sestest-win10-x64.zip
+    }
+}
+
+Task Publish.SesTest.Linux64 -Depends Requires.DotNetExe, Restore.NuGetPackages {
+    exec {
+        & $dotnetExe publish $srcDir\sestest\sestest.csproj --self-contained --output $publishDir\sestest\linux-x64 --runtime linux-x64
+    }
+
+    exec {
+        compress-archive $publishDir\sestest\linux-x64\* $publishDir\sestest-linux-x64.zip
+    }
+}
+
+## --------------------------------------------------------------------------------
+##   Configuration Targets
+## --------------------------------------------------------------------------------
+## Tasks used to configure the build
+
+Task Request.x64 {
+    $script:targetPlatform = "x64"
+}
+
+Task Request.x86 {
+    $script:targetPlatform = "x86"
+}
+
+Task Request.Debug {
+    $script:configuration = "Debug"
+}
+
+Task Request.Release {
+    $script:configuration = "Release"
+}
+
+## --------------------------------------------------------------------------------
+##   Prerequisite Targets
+## --------------------------------------------------------------------------------
+## Tasks used to ensure that prerequisites are available when needed. 
+
+Task Requires.Configuration {
+    if ($configuration -eq $null) {
+        $script:configuration = "Debug"
+    }
+
+    Write-Host "Build configuration is $configuration"
+}
+
+Task Requires.DotNetExe {
+    $script:dotnetExe = (get-command dotnet).Path
+
+    if ($dotnetExe -eq $null)
+    {
+        $script:dotnetExe = resolve-path $env:ProgramFiles\dotnet\dotnet.exe -ErrorAction SilentlyContinue
+    }
+    
+    if ($dotnetExe -eq $null)
+    {
+        throw "Failed to find dotnet.exe"
+    }
+
+    Write-Host "Dotnet executable: $dotnetExe"
+}
+
+Task Requires.MsBuild {
+    # prefer MSBuild from VS2017 if its there
+    $script:msbuildExe = resolve-path "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2017\*\MSBuild\*\Bin\MSBuild.exe" -ErrorAction SilentlyContinue
+
+    if ($msbuildExe -eq $null)
```

Comment:
Do you need to qualify this with `script:` too? (same with `dotnetExe` twice above?) Or does it only matter when writing to the variables?

Reply:
The scope `script:` is only needed when writing to the variables. Reading automagically walks through progressively larger scopes until a value is found.